src/components/inputs/number/number.component.tsx

```diff
@@ -11,6 +11,28 @@ import { useFormProviderContext } from '../../../provider/form-provider';
 import FieldLabel from '../../field-label/field-label.component';
 import { isEmpty } from '../../../validators/form-validator';
 
+
+const extractFieldUnitsAndRange = (concept) => {
```

Comment:
Can you wrap this in a [useCallback](https://react.dev/reference/react/useCallback)?

Reply:
No, actually, you shouldn't do that here. This is actually done correctly. Functions defined inside functional components would be swapped with a new instance every time the function is called (so, every time the renderer detects that the component might change). `useCallback()` prevents that by ensuring the variable keeps a stable reference to the same function until the dependency array changes. Since this function isn't defined in a functional component body, references to this function will always be stable references to the same function, so `useCallback()` isn't needed here (and, IMO, we should prefer these ancillary functions be defined outside of functional components wherever feasible).

All that said, the `concept` argument here must have a type defined.